plugins/backend/catalog-backend-module-aws/src/providers/AWSEBSVolumeProvider.ts

```diff
@@ -0,0 +1,161 @@
+import { ANNOTATION_VIEW_URL, ResourceEntity } from '@backstage/catalog-model';
+import { LoggerService } from '@backstage/backend-plugin-api';
+import * as winston from 'winston';
+import { EC2 } from '@aws-sdk/client-ec2';
+import { AWSEntityProvider } from './AWSEntityProvider';
+import {
+  LabelValueMapper,
+  ownerFromTags,
+  relationshipsFromTags,
+} from '../utils/tags';
+import { DynamicAccountConfig } from '../types';
+import { duration } from '../utils/timer';
+import { DescribeVolumesCommandOutput } from '@aws-sdk/client-ec2/dist-types/commands/DescribeVolumesCommand';
+
+const ANNOTATION_EBS_VOLUME_ID = 'amazonaws.com/ebs-volume-id';
+
+const createEbsVolumeConsoleLink = (region: string, volumeId: string): string =>
+  `https://${region}.console.aws.amazon.com/ec2/home?region=${region}#VolumeDetails:volumeId=${volumeId}`;
+
+/**
+ * Provides entities from AWS Elastic Block Store volumes.
+ */
+export class AWSEBSVolumeProvider extends AWSEntityProvider {
+  static fromConfig(
+    config: Config,
+    options: {
+      logger: winston.Logger | LoggerService;
+      catalogApi?: CatalogApi;
+      providerId?: string;
+      ownerTag?: string;
+      useTemporaryCredentials?: boolean;
+      labelValueMapper?: LabelValueMapper;
+    },
+  ) {
+    const accountId = config.getString('accountId');
+    const roleName = config.getString('roleName');
+    const roleArn = config.getOptionalString('roleArn');
+    const externalId = config.getOptionalString('externalId');
+    const region = config.getString('region');
+
+    return new AWSEBSVolumeProvider(
+      { accountId, roleName, roleArn, externalId, region },
+      options,
+    );
+  }
+
+  getProviderName(): string {
+    return `aws-ebs-volume-provider-${this.providerId ?? 0}`;
+  }
+
+  private async getEc2(dynamicAccountConfig?: DynamicAccountConfig) {
+    const { region } = this.getParsedConfig(dynamicAccountConfig);
+    const credentials = this.useTemporaryCredentials
+      ? this.getCredentials(dynamicAccountConfig)
+      : await this.getCredentialsProvider();
+    return this.useTemporaryCredentials
+      ? new EC2({ credentials, region: region })
+      : new EC2(credentials);
+  }
+
+  async run(dynamicAccountConfig?: DynamicAccountConfig): Promise<void> {
+    if (!this.connection) {
+      throw new Error('Not initialized');
+    }
+    const startTimestamp = process.hrtime();
+
+    const { region, accountId } = this.getParsedConfig(dynamicAccountConfig);
+    const groups = await this.getGroups();
+
+    this.logger.info(`Providing EBS volume resources from aws: ${accountId}`);
+    const ebsResources: ResourceEntity[] = [];
+
+    const ec2 = await this.getEc2(dynamicAccountConfig);
+    const defaultAnnotations = await this.buildDefaultAnnotations(
+      dynamicAccountConfig,
+    );
+
+    let nextToken: string | undefined = undefined;
+    do {
+      const volumes: DescribeVolumesCommandOutput = await ec2.describeVolumes({
```

Comment:
The aws-sdk exposes a pagination api..

const paginatorConfig = {
  client: ec2,
  pageSize: 25
};
const commandParams = {};
const paginator = paginateDescribeVolumes(paginatorConfig, commandParams);

const volumes = [];
for await (const page of paginator) {
  // page contains a single paginated output.
  volumes.push(...page.Volumes);
}

You dont have to do it yourself.